samples/contrib/azure-samples/kfp-azure-databricks/tests/test_run_op.py

```diff
@@ -0,0 +1,494 @@
+import unittest
+from pathlib import Path
+import kfp
+from kfp.dsl import PipelineParam
+from databricks import SubmitRunOp, DeleteRunOp
+
+class TestSubmitRunOp(unittest.TestCase):
+
+    def test_databricks_submit_run_without_k8s_or_run_name(self):
+        def my_pipeline():
+            SubmitRunOp(
+                name="submitrun",
+                new_cluster={
+                    "spark_version":"5.3.x-scala2.11",
+                    "node_type_id": "Standard_D3_v2",
+                    "num_workers": 2
+                },
+                libraries=[
+                    {"jar": "dbfs:/my-jar.jar"},
+                    {"maven": {"coordinates": "org.jsoup:jsoup:1.7.2"}}
+                ],
+                spark_jar_task={
+                    "main_class_name": "com.databricks.ComputeModels"
+                }
+            )
+
+        self.assertRaises(ValueError, lambda: kfp.compiler.Compiler()._compile(my_pipeline))
+
+    def test_databricks_submit_run_with_job_name_and_jar_params(self):
+        def my_pipeline():
+            run_name = "test-run"
+            job_name = "test-job"
+            jar_params = ["param1", "param2"]
+
+            expected_spec = {
+                "run_name": run_name,
+                "job_name": job_name,
+                "jar_params": jar_params
+            }
+
+            res = SubmitRunOp(
+                name="submitrun",
+                run_name=run_name,
+                job_name=job_name,
+                jar_params=jar_params
+            )
+
+            self.assert_res(res, expected_spec)
+
+        kfp.compiler.Compiler()._compile(my_pipeline)
+
+    def test_databricks_submit_run_with_job_name_and_python_params(self):
+        def my_pipeline():
+            run_name = "test-run"
+            job_name = "test-job"
+            python_params = ["john doe", "35"]
+
+            expected_spec = {
+                "run_name": run_name,
+                "job_name": job_name,
+                "python_params": python_params
+            }
+
+            res = SubmitRunOp(
+                name="submitrun",
+                run_name=run_name,
+                job_name=job_name,
+                python_params=python_params
+            )
+
+            self.assert_res(res, expected_spec)
```

Comment:
Maybe silly question, where was `assert_res` defined?

Reply:
It is defined in that same class. Line 413

Reply:
Thanks!